sdk/communication/Azure.Communication.Administration/src/CommunicationIdentityClient.cs

```diff
@@ -29,16 +29,34 @@ public CommunicationIdentityClient(string connectionString, CommunicationIdentit
                   options ?? new CommunicationIdentityClientOptions(),
                   ConnectionString.Parse(AssertNotNull(connectionString, nameof(connectionString))))
         { }
-
+        /// <summary> Initializes a new instance of <see cref="CommunicationIdentityClient"/>.</summary>
+        /// <param name="tokenCredential">Token acquired from the Azure Communication Services to be used by service calls</param>
+        /// <param name="endpoint">Endpoint to hit on the server</param>
+        /// <param name="options">Client option exposing <see cref="ClientOptions.Diagnostics"/>, <see cref="ClientOptions.Retry"/>, <see cref="ClientOptions.Transport"/>, etc.</param>
+        public CommunicationIdentityClient(Uri endpoint, TokenCredential tokenCredential, CommunicationIdentityClientOptions? options = default)
+            : this(
+                AssertNotNull(endpoint, nameof(endpoint)),
+                options ?? new CommunicationIdentityClientOptions(),
+                AssertNotNull(tokenCredential, nameof(tokenCredential)))
+        { }
         private CommunicationIdentityClient(CommunicationIdentityClientOptions options, ConnectionString connectionString)
         {
             _clientDiagnostics = new ClientDiagnostics(options);
             RestClient = new CommunicationIdentityRestClient(
                 _clientDiagnostics,
-                options.BuildHttpPipline(connectionString),
+                options.BuildHttpPipeline(connectionString),
                 connectionString.GetRequired("endpoint"));
         }
 
+        private CommunicationIdentityClient(Uri endpoint, CommunicationIdentityClientOptions options, TokenCredential tokenCredential)
```

Comment:
nit: even though it's private it reads better if options are the last argument

Reply:
I don't think I can make this change. The method signature will conflict with the public one and throw a compiler error.